src/lib.rs

```diff
@@ -819,19 +819,20 @@ macro_rules! checked_arith_impl {
         impl<T: Clone + Integer + CheckedMul + $imp> $imp for Ratio<T> {
             #[inline]
             fn $method(&self, rhs: &Ratio<T>) -> Option<Ratio<T>> {
-                let ad = otry!(self.numer.checked_mul(&rhs.denom));
-                let bc = otry!(self.denom.checked_mul(&rhs.numer));
-                let bd = otry!(self.denom.checked_mul(&rhs.denom));
-                Some(Ratio::new(otry!(ad.$method(&bc)), bd))
+                let gcd = self.denom.clone().gcd(&rhs.denom.clone());
```

Comment:
It's possible for `gcd` to overflow and panic, particularly if both denominators are signed `MIN`, since `gcd` always returns positive values.
https://github.com/rust-num/num-integer/blob/bcce4ccbb293e3c6eee0098a1f74a0479739a48c/src/lib.rs#L475-L480

I'm hesitant whether this is worth caring about, since I believe the only way to create a `Ratio` with a negative denominator at all is with `new_raw` -- but that does exist and is a safe function. Since this is only a problem for checked ops, maybe it is worth being more careful.

An equal-denominator case, as I suggested on the unchecked ops, would solve the basic signed-`MIN` problem. I suppose some other generic type has unknown edge cases here, but I can't think of a way to deal with that without adding additional `Checked*` traits constraining this `impl`. (e.g. `CheckedSub` would allow a manual checked-`abs()`). But it's probably good enough to just solve signed `MIN`.

Returning `Ratio::new` here will fail if the final reduced denominator is still `MIN`, but checking for equal denominators is a reasonable optimization anyway.

Reply:
I think we need to spell out exactly what are and aren't valid rational numbers in this library, before we can answer questions like these. see my comments on #8